### PR TITLE
Add retries

### PR DESCRIPTION
**Pull request by hatchan**

### CODE REVIEW

**Comment by mies** [comment]
looks good but is retry optional/configured with an option? if so might be good to update the readme/instructions as well

**Comment by hatchan** [comment]
Currently it will retry 3 times. Didn't think it would be necessary to make this configurable. Will make note  of the retries in the README though.

For users where the step succeeds, the retry won't be hit.
For users where the step keeps failing, it will fail 3 times, and then exits.
For users where it sometimes fails, they will be saved by the retry.

Let's see if users really care about making it configurable.

File: run.sh [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+}
+
+npm_install() {
+  local retries=3;
+  for try in $(seq "$retries"); do
+    info "Starting npm install, try: $try"
+    npm install $WERCKER_NPM_INSTALL_OPTIONS && return;
+  done
+
+  fail "Unable to install npm, retries: $retries"
```
Comment (flenter) [inline run.sh]: Maybe change this to `Failed to successfully execute npm install , retries: $retries`. It now sounds to me like we were trying to install `npm` itself.

**Comment by flenter** [comment]
minor comment, lgtm!

File: run.sh (hunk 1 above)
Comment (hatchan) [inline run.sh]: Good one

**Comment by yoshuawuyts** [comment]
We might want to do this one in POSIX sh rather than bash since there are less guarantees about the system this will be running on. Other than that LGTM.

**Comment by hatchan** [comment]
@yoshuawuyts all code is POSIX compliant.

**Pull request merged**